File: test_models.py
import os
import requests

api_key = None
with open(".env", "r") as f:
    for line in f:
        if line.startswith("GEMINI_API_KEY="):
            api_key = line.strip().split("=", 1)[1].strip('"\'')

print("Fetching models...")
response = requests.get(f"https://generativelanguage.googleapis.com/v1alpha/models?key={api_key}")
data = response.json()
if "models" in data:
    for m in data["models"]:
        methods = m.get("supportedGenerationMethods", [])
        if "bidiGenerateContent" in methods:
            print(f"Supported model: {m['name']}")
else:
    print(data)

File: test_models.js
const https = require('https');

const API_KEY = process.env.GEMINI_API_KEY || 'your-api-key-here'; // We will replace this with their actual key from test_live.py

https.get(`https://generativelanguage.googleapis.com/v1beta/models?key=${API_KEY}`, (resp) => {
  let data = '';
  resp.on('data', (chunk) => { data += chunk; });
  resp.on('end', () => {
    try {
      const json = JSON.parse(data);
      if (json.models) {
        const bidiModels = json.models.filter(m => 
          m.supportedGenerationMethods && m.supportedGenerationMethods.includes('bidiGenerateContent')
        );
        console.log("Models supporting bidiGenerateContent:");
        bidiModels.forEach(m => console.log(m.name));
      } else {
        console.log("No models array found.", json);
      }
    } catch (e) {
      console.log("Error parsing JSON:", e.message);
    }
  });
}).on("error", (err) => {
  console.log("Error: " + err.message);
});
